src/clients/Deluge.ts

```diff
@@ -22,11 +22,23 @@ enum DelugeErrorCode {
 	RPC_FAIL = 4,
 	BAD_JSON = 5,
 }
-interface TorrentInfo {
-	complete: boolean;
-	save_path: string;
-	label?: string;
-}
+type injectData = [
+	string,
+	string,
+	{
+		add_paused: boolean;
+		seed_mode: boolean;
+		download_location: string;
+	}
+];
+
+type DelugeJSON<DelugeResult, ErrorType> = {
+	result?: DelugeResult;
+	error?: ErrorType;
+};
+type ErrorType = { message?: string; code?: DelugeErrorCode };
+type DelugeResult = boolean | string | TorrentStatus;
+type TorrentStatus = { torrents?: Record<string, TorrentInfo> };
```

Comment:
here's what I would do
```ts
// this one's only relevant for getTorrentInfo I think
type TorrentStatus = { torrents?: Record<string, TorrentInfo> };

// we don't know which one of these it's gonna be, it depends on which method we called
type DelugeResult = boolean | string | TorrentStatus;

// errors always have the same shape regardless of which method we called
type ErrorType = { message?: string; code?: DelugeErrorCode };

// DelugeJSON can represent any number of responses based on which method we call
// so it has to take a type argument ResultType which the caller will provide.
type DelugeJSON<ResultType> = {
	result?: ResultType;

// since errors always have the same shape, 
	error?: ErrorType;
};

...

// implementing call (ResultType is a type variable that we then feed into the same-named ResultType of DelugeJSON)
call<ResultType>(method: string, params: Whatever): Promise<DelugeJSON<ResultType>>

...

// calling call inside getTorrentInfo
this.call<TorrentStatus>("core.get_torrent_status", ...);

// calling call inside inject
this.call<boolean>("core.add_torrent", ...);
```